:memo: Doc-A-Thon Blog Post (#670)
ros-planning/moveit.ros.org: _posts/2022-01-20-doc-a-thon.md

---
author: Tyler Weaver
comments: false
date: 2022-01-20 17:00:00+07:00
layout: post
slug: 2022-doc-a-thon

> title: 2022 Doc-A-Thon

Reviewer: ```suggestion
title: 2022 MoveIt Doc-A-Thon
```

description: 2022 Jan 14
media_type: image
media_link: /assets/images/blog_posts/2022_01_22_doc_a_thon.png
categories:
- MoveIt
- ROS
---

## 2022 Jan 14

Last Friday the 14th, PickNik had an internal day of working on the documentation site for [MoveIt 2](https://moveit.picknik.ai/).

> There are a few really large improvements and efforts that were have begun.

Reviewer: ```suggestion
There are a few really large improvements and efforts that were have begun.
```
```suggestion
There are a few really large improvements and efforts that we tackled.
```

- [livanov93](https://github.com/livanov93) added the configuration and a [How To Generate API Doxygen Reference Locally](https://moveit.picknik.ai/galactic/doc/how_to_guides/how_to_generate_api_doxygen_locally.html) page on the docs site.
- [vatanaksoytezer](https://github.com/vatanaksoytezer) did the work to make it deploy automatically using GitHub actions as part of the site.
- [senceryazici](https://github.com/senceryazici) added a CSS stylesheet to add a clean modern style to the Doxygen site.

### Site Content Redesign

Following the advice of [The Documentation System](https://documentation.divio.com/) we separated the different types of documentation.
This redesign should make it easy for those using the site to find the sort of content they are looking for.
- [Tutorials](https://moveit.picknik.ai/galactic/doc/tutorials/tutorials.html) are *learning-oriented* and guide the user through creating projects using MoveIt to explore the various features MoveIt offers.
- [How-To Guides](https://moveit.picknik.ai/galactic/doc/how_to_guides/how_to_guides.html) are *problem-oriented* in that they are focused on helping the reader solve a specific problem.

> - [Concepts](https://moveit.picknik.ai/galactic/doc/concepts/concepts.html) are *understanding-oriented* and should help those trying to get a deeper understanding to why questions about MoveIt.

Reviewer: ```suggestion
- [Concepts](https://moveit.picknik.ai/galactic/doc/concepts/concepts.html) are *understanding-oriented* and should help those trying to get a deeper understanding to "why" questions about MoveIt.
```

> - [API Documentation](https://moveit.picknik.ai/galactic/api/html/index.html) is *information-oriented* and serves to help you find specific answers about the API of MoveIt.

Reviewer: ```suggestion
- [API Documentation](https://moveit.picknik.ai/galactic/api/html/index.html) is *information-oriented* and serves to help you find specific answers about MoveIt's API.
```

- [Examples](https://moveit.picknik.ai/galactic/doc/examples/examples.html) are *feature-oriented*.  These contain examples of how to use specific features in MoveIt.  The majority of the content on the site is here because this was the previous format for tutorials.
- [Contributing](https://moveit.picknik.ai/galactic/doc/how_to_contribute/how_to_contribute.html) docs are *contributor-oriented*.  This is the place for documenting the contribution process and concepts that are specifically useful when trying to make changes to MoveIt.

### Look and Feel

> Previous to this day of improvements, we upgraded the infrastructure of the site to a feature that allows us to deploy different versions of the site for the different releases of MoveIt.
> One thing that was lost in this change was the look and feel of the site that was carefully designed to use the same color scheme as the moveit.ros.org website.
> [Dave](https://github.com/davetcoleman) restored the look and feel of the site and also made other small changes to make it a much nicer site.

Reviewer: ```suggestion
Before this doc-a-thon, we had upgraded the infrastructure of the site to allow us to deploy different versions of the site for the different releases of MoveIt.
With that change, we lost the look and feel of the original site, which had been carefully designed to use the same color scheme as the moveit.ros.org website.
[Dave](https://github.com/davetcoleman) restored the look and feel of the site and also made other small changes to improve the user experience.
```
